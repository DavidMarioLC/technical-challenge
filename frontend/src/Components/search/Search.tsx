import "./search.css";
import {IRepository} from '../../interfaces/repository.interfaces';
import { useEffect, useState } from "react";
interface Props {
  valueSearch:string;
  repositories:IRepository[];
  setRepositoriesSearch:(value:IRepository[]) => void;
  setValueSearch:(value:string) => void;
}

const Search = ({setValueSearch,setRepositoriesSearch,repositories,valueSearch}: Props) => {

 
  
  const filterRepositories = (value:string) =>{ 
    const NewListRepositories = repositories.filter((repo) => repo.name.toLowerCase().includes(value))
    setRepositoriesSearch(NewListRepositories)
  }

  const handlerInput = (e:any) => {
    setValueSearch(e.target.value)
    filterRepositories(e.target.value)
  }



  



  return (
    <div className="search">
        <input
        value={valueSearch}
        onChange={handlerInput}
        placeholder="Find a repositoryâ€¦"
        className="search__input"
        type="search" />
    </div>
  )
}

export default Search
