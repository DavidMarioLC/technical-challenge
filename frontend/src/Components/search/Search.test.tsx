import renderer from 'react-test-renderer';
import Search from './Search';

it('render Search correctly', () => {
  const tree = renderer.create(<Search setValueSearch={()=>{}} valueSearch={''} />).toJSON();
  expect(tree).toMatchSnapshot();
});