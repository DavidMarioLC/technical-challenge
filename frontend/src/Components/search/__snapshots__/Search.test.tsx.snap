// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Search correctly 1`] = `
<div
  className="search"
>
  <input
    className="search__input"
    onChange={[Function]}
    placeholder="Find a repositoryâ€¦"
    type="search"
    value=""
  />
</div>
`;
